data/creaturescripts/scripts/boss.lua

```diff
@@ -0,0 +1,229 @@
+			table.insert(buffer, ";")
+
+			if total ~= 0 then
+				db.query(table.concat(buffer))
+			end
+		end
+	)
+end
+
+local function getPlayerStats(bossId, playerGuid, autocreate)
+	local ret = globalBosses[bossId][playerGuid]
+	if not ret and autocreate then
+		ret = {
+			bossId = bossId, 
+			damageIn = 0, -- damage taken from the boss
+			healing = 0, -- healing (other players) done
+		}
+		globalBosses[bossId][playerGuid] = ret
+		return ret
+	end
+	return ret
+end
+
+function onDeath(creature, corpse, killer, mostDamageKiller, lastHitUnjustified, mostDamageUnjustified)
+	local monsterType = creature:getType()
+	if monsterType:isRewardBoss() then -- Make sure it is a boss
+		local bossId = creature:getId()
+		local timestamp = os.time()
+
+		local totalDamageOut, totalDamageIn, totalHealing = 0.1, 0.1, 0.1 -- avoid dividing by zero
+
+		local scores = {}
+		local info = globalBosses[bossId]
+		local damageMap = creature:getDamageMap()
+
+		for guid, stats in pairs(info) do
+			local player = Player(stats.playerId)
+			local part = damageMap[stats.playerId]
+			local damageOut, damageIn, healing = (stats.damageOut or 0) + (part and part.total or 0), stats.damageIn or 0, stats.healing or 0
+
+			totalDamageOut = totalDamageOut + damageOut
+			totalDamageIn = totalDamageIn + damageIn
+			totalHealing = totalHealing + healing
+
+			table.insert(scores, {
+				player = player, 
+				guid = guid,
+				damageOut = damageOut, 
+				damageIn = damageIn,
+				healing = healing,
+			})            
+		end
+
+		local participants = 0
+		for _, con in ipairs(scores) do
+			local score = (con.damageOut / totalDamageOut) + (con.damageIn / totalDamageIn) + (con.healing / totalHealing)
+			con.score = score / 3 -- normalize to 0-1
+			if score ~= 0 then
+				participants = participants + 1
+			end
+		end
+		table.sort(scores, function(a, b) return a.score > b.score end)
+
+		local expectedScore = 1 / participants
+
+		for _, con in ipairs(scores) do
+			local reward, stamina -- ignoring stamina for now because I heard you receive rewards even when it's depleted   
+			if con.player then   
+				reward = con.player:getReward(timestamp, true)
+				stamina = con.player:getStamina()
+			else
+				stamina = con.stamina or 0
+			end
+
+			local playerLoot
+			if --[[stamina > 840 and]] con.score ~= 0 then
+				local lootFactor = 1
+				lootFactor = lootFactor / participants ^ (1 / 3) -- tone down the loot a notch if there are many participants
+				lootFactor = lootFactor * (1 + lootFactor) ^ (con.score / expectedScore) -- increase the loot multiplicatively by how many times the player surpassed the expected score
+				playerLoot = monsterType:getBossReward(lootFactor, _ == 1)
+
+				if con.player then
+					for _, p in ipairs(playerLoot) do
+						reward:addItem(p[1], p[2])
+					end
+				end
+			end
+
+			if con.player then
+				local lootMessage = {"The following items are available in your reward chest: "}
+
+				if --[[stamina > 840]]true then
```

Comment:
In that case add a comment above, and id comment out the if statment so when we know about it we can remove the comments.
One less thing to load for now.